### PR TITLE
3x tweaks

### PR DESCRIPTION
**Pull request by manolama**

### CODE REVIEW

File: core/src/main/java/net/opentsdb/query/processor/downsample/DownsampleConfig.java [hunk 1]
```diff
@@ -279,6 +309,15 @@ public int intervals() {
     return cached_intervals;
   }
   
+  /**
+   * Returns the auto intervals for this factory.
+   * <b>WARNING:</b> Do NOT modify the list or entries.
+   * @return The non-null intervals list.
+   */
+  public List<Pair<Long, String>> autoIntervals() {
+    return intervals;
```
Comment (SeanPMiller) [inline core/src/main/java/net/opentsdb/query/processor/downsample/DownsampleConfig.java]: You could ensure no modification by returning an immutable view.

**Comment by SeanPMiller** [comment]
Per discussion, removing all unneeded files from the old build system is a good idea.

File: core/src/main/java/net/opentsdb/query/processor/downsample/DownsampleFactory.java [hunk 2]
```diff
@@ -108,6 +229,154 @@ public void setupGraph(final TimeSeriesQuery query,
...
+            pairs.add(new Pair<Long, String>(interval, entry.getValue()));
+          } catch (Exception e) {
+            LOG.error("Failed to parse entry: " + entry + ". Using defaults", e);
+            return;
+          }
+        }
+        
+        // all good, sync it. Ugly yeah but this should run infrequently and only
+        // have up to a dozen entries.
+        synchronized (intervals) {
```
Comment (SeanPMiller) [inline core/src/main/java/net/opentsdb/query/processor/downsample/DownsampleFactory.java]: Seeing `synchronized` draws my attention. Investigated. Don't see any big problems here...

**Comment by manolama** [comment]
Merged in 2ee4d3ada1cca88ea3e0a3ab14390ba1bfb223bf

**Pull request closed without merging**